src/app.ts

```diff
@@ -134,6 +137,50 @@ const registerMiddleware = (app: Express): void => {
     }
     next()
   })
+
+  // Access Control List
+  app.use(async (req: Request, res: Response, next: NextFunction) => {
+    const postWhitelist = [
+      Routes.CONTACT,
+      Routes.LOGIN,
+      Routes.REGISTER,
+      Routes.RESET_GUEST_PASSWORD
+    ]
+    const user = req.user
+    const url = req.url as Routes
+    const method = req.method.toLowerCase()
+
+    if (!user && 'post' === method && postWhitelist.includes(url)) {
+      return next()
+    }
+
+    // ex: ["user", "admin"]
+    const currentUserRoles = user?.userRoles.map((userRole) => userRole.role.name)
+
+    if (!currentUserRoles) {
+      return res.sendStatus(401)
+    }
+
+    try {
+      // TODO possibly only read the file once
+      const aclRoles = yaml.safeLoad(fs.readFileSync('acl.yaml', 'utf-8')) as Role[]
+
+      const hasPermissions = aclRoles.filter((value: Role) => currentUserRoles.indexOf(value.name) > -1
+        && method in value
+        && value[method].includes(url)
+        ).length > 0
+
+      if (!hasPermissions) {
+        return res.sendStatus(401)
+      }
+
+    } catch (error) {
+      logger.error(error)
+      return res.sendStatus(500).json({ error: 'something went wrong' })
```

Comment:
You don't want to use `sendStatus`, we want to use `status`. Also I've since updated this to be `message` instead of `error` and updated the casing a little:
```suggestion
      return res.status(500).json({ message: 'Something went wrong' })
```
Again I plan on having these all in a central place so we don't need to maintain remembering silly string values and object keys.

Reply:
Now that we don't need to read a file, I'm removing the try/catch so I'm deleting this line